kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/preprocess/SinkRecordFilterTest.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    @Test
+    public void testExistJsonPath(){
+
+        // test when there's at least one node that satisfies condition
+        sinkRecordFilter = initializeSinkRecordFilter(INCLUDE_CONDITION_CONFIG,
+                "$.Inventories[?(!!@..FINANCIAL_BLOCK)]");
+
+        assertEquals(sinkRecordFilter.getConfigType(), INCLUDE_CONDITION_CONFIG);
+        assertEquals(sinkRecordFilter.getFilterFields(testMap).size(), 2);
+        assertEquals(sinkRecordFilter.satisfyCondition(testMap), true);
+
+        sinkRecordFilter = initializeSinkRecordFilter(INCLUDE_CONDITION_CONFIG,
+                "$[?((@.Inventories[0].COUNTRY_ID == 8 && @.Inventories[1].COUNTRY_ID < 8) || " +
+                        "@['High Performance Mixed Signal ( HPMS )'][0].COUNTRY_ID == 8)]");
+
+        assertEquals(sinkRecordFilter.getConfigType(), INCLUDE_CONDITION_CONFIG);
+        assertEquals(sinkRecordFilter.getFilterFields(testMap).size(), 1);
+        assertEquals(sinkRecordFilter.satisfyCondition(testMap), true);
+
+        // test when node does not satisfies condition
+        sinkRecordFilter = initializeSinkRecordFilter(INCLUDE_CONDITION_CONFIG,
+                "$.Inventories[?(@..FINANCIAL_BLOCK == 'Balance Sheet')]");
+
+        assertEquals(sinkRecordFilter.getConfigType(), INCLUDE_CONDITION_CONFIG);
+        assertEquals(sinkRecordFilter.getFilterFields(testMap).size(), 0);
+        assertEquals(sinkRecordFilter.satisfyCondition(testMap), false);
+
+        sinkRecordFilter = initializeSinkRecordFilter(INCLUDE_CONDITION_CONFIG,
+                "$['High Performance Mixed Signal ( HPMS )'][0][?(@.FINANCIAL_BLOCK == 'Balance Sheet')]");
+
+        assertEquals(sinkRecordFilter.getConfigType(), INCLUDE_CONDITION_CONFIG);
+        assertEquals(sinkRecordFilter.getFilterFields(testMap).size(), 0);
+        assertEquals(sinkRecordFilter.satisfyCondition(testMap), false);
+
+        // test when only path is given, no condition(expected to return nodes corresponding to JsonPath)
+        sinkRecordFilter = initializeSinkRecordFilter(INCLUDE_CONDITION_CONFIG,
+                "$.Inventories[?(@.FINANCIAL_BLOCK)]");
+
+        assertEquals(sinkRecordFilter.getConfigType(), INCLUDE_CONDITION_CONFIG);
+        assertEquals(sinkRecordFilter.getFilterFields(testMap).size(), 2);
+        assertEquals(sinkRecordFilter.satisfyCondition(testMap), true);
+
+    }
+
+    @Test
+    public void testNotExistJsonPath(){
+        sinkRecordFilter = initializeSinkRecordFilter(INCLUDE_CONDITION_CONFIG,
+                "$.Inventories[0].nonexitentField");
+
+        final TestAppender appender = new TestAppender();
+        final Logger rootLogger = Logger.getRootLogger();
+
+        rootLogger.addAppender(appender);
+
+        List filtered = sinkRecordFilter.getFilterFields(testMap);
+
+        assertEquals(filtered.size(), 0);
+
+        assertTrue(assertLogged(CoreMatchers.containsString(
+                "config can not be found in json object"), appender));
```

Comment:
testing for log statements can be brittle, so I would avoid it unless there is no other option to check for a result. Here you already checked that the filtered size size is 0, so that should be enough, no?